courses/COMP-2310/surviving2310.md

```diff
@@ -0,0 +1,62 @@
+---
+id: surviving2310
+title: How To Survive COMP-2310
+sidebar_label: Surviving COMP-2310
+slug: /COMP-2310/survivingCOMP2310
+---
+
+Chances are you've probably heard a lot about COMP-2310 from your peers (and if not, you're in for a fun time). COMP-2310, Theoretical Foundations of Computer Science, is typically one of the most difficult courses that undergraduate CS students take during their degree. It is a comprehensive, rigorous course which focuses heavily on mathematical proofs. A higher level of rigour is expected in this class when compared to COMP-1000. Assignments and tests often have an average well below 50%. Simply put, if you don't take this course seriously, you can and will fail.
+
+## How to Succeed in This Course
+
+The good news is that if you take this course seriously and put in the work, passing (and even performing well) is competely attainable. Here are some ways to better your chances of doing well in this course:
+
+### Purchase (and Read) the Courseware
+
+Normally with CS classes the textbook is optional (and not helpful). COMP-2310 is an exception to this rule, however. All of the lecture content, practice questions, and assignment questions are contained within this book. Moreover, in this class you're expected to use a very specific form and nomenclature in proofs. This form and nomenclature is unique to this course, thus any terminology or rules you've learned in other courses will no longer be useful or valid. For example, Modus Tollens:
+
+```
+(P => Q), ~Q
+================
+~P
+```
+
+is referred to in COMP-2310 as "I4". Without the textbook, there'd be no way of knowing that this difference exists. (And, as you may expect, using the name "Modus Tollens" rather than "I4" WILL result in you getting a 0 on that question).
+
+### Go to the Labs
+
+During the labs, you'll have the chance to go over actual practice questions together as a group. This will be incredibly helpful as this will likely be your first time seeing how to solve questions like these.
+
+In addition to the educational benefit you'll get from attending the labs, you're also more likely to gain favour from the professor. While the labs may not be worth any concrete marks, the professor does take attendance, and not showing up may end up affecting your final mark. This is because there's usually a very large curve in this class for your final mark. Unlike other classes though, the curve is not always applied the same to all students. Students who show that they're trying are more likely to get a higher curve, and thus a better mark.
```

Comment:
```suggestion
In addition to the educational benefit you'll get from attending the labs, you're also more likely to gain favour from the professor. While the labs may not be worth any concrete marks, the professor does take attendance, and not showing up will end up affecting your final mark. This is because there's usually a very large curve in this class for your final mark. Unlike other classes though, the curve is not always applied the same to all students. Students who show that they're making their best effort, and always attend and participate in labs, are likely to get a higher curve, and thus a better mark.
```